src/gradualizer.erl

```diff
@@ -6,6 +6,12 @@
 %%%   in the given directory.
 %%% - `print_file': if `true' prefix error printouts with the file name the
 %%%   error is from.
+%%% - `crash_on_error`: if `true` crash on the first produced error
```

Comment:
Perhaps this option should be called `crash_on_first_error`.
Also, I think the word "crash" might be a bit imprecise. A suggestion would be "exception" which is more accurate.

Reply:
I'm not introducing this option, it was just not documented. I can split this change into a separate PR that also includes renaming it.

Reply:
I added it for debugging purposes some months ago. I wasn't sure if it would be better to leave it undocumented. I didn't include the word `first` because there is obviously only one crash... By crash, I meant the same behaviour as for an unexpected programming error (case_clause, etc.). I needed the stack trace.

I don't know if exception is a better name. Less precise perhaps. An exception (throw/error/exit) can be caught and the checking can continue, as we normally do with type errors.

Perhaps we can delete `crash_on_error` when the typechecker is becoming more stable. I don't think it is stable enough yet though. I have seen gradualizer hang, so yesterday (unpushed) I added a `debug` option for printing progress stuff like "Checking X :: Y" while debugging, to be able to locate where it hangs.

The point is that it might be good to have options for debugging the gradualizer and these should probably be either undocumented or marked in the documentation as "you probably don't want to use this; it's for debugging the Gradualizer".

Reply:
Should I just remove this comment then and instead add a comment in `typechecker`  about removing the option when it's no longer needed?

Reply:
I don't care.

(It's good to avoid unrelated changes in the same PR. People tend to comment on them. :-))

Reply:
Feel free to resolve this conversation. I agree that changing the option would be an unrelated change and we shouldn't do it as part of this PR.